pkg/controller/machineset/controller.go

```diff
@@ -265,6 +265,10 @@ func (r *ReconcileMachineSet) reconcile(ctx context.Context, machineSet *machine
 		return reconcile.Result{}, errors.Wrap(err, "failed to update machine set status")
 	}
 
+	if syncErr != nil {
```

Comment:
Why do we only handle this after the `updateMachineSetStatus` bit? Are those always tied together this way? If so, should that just be a part of `syncReplicas`?

Reply:
MachinesetStatus and events are two mechanisms through which user can get to know about current state. Even if syncReplicas is facing error, i thought it would help if we let the stats updated in the machinesetstatus.

Reply:
From a purely pattern perspective, why do we not check for a `syncErr` straight after we call:

```go
syncErr := r.syncReplicas(machineSet, filteredMachines)
```

Reply:
If calling `syncReplicas` can fail why do we not fail fast when this is true?

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/880#discussion_r273421136

Reply:
Discussed this offline. I think it might make sense to just move the `updateMachineSetStatus` call to `syncReplicas` if we always want to update the status while syncing.  Not a big deal though, and this merged as-is upstream, so LGTM.

Reply:
> [kubernetes-sigs#880 (comment)](https://github.com/kubernetes-sigs/cluster-api/pull/880#discussion_r273421136)

In which case I would say add those exact words as to why we delay checking for success/failure on this call.

Reply:
> so that machineset status status could get updated and along with error message in the events, user could have latest stats.